app/views/catalog/record/_mods_bibliographic.html.erb

```diff
@@ -24,6 +24,6 @@
   <%= render ModsDisplay::FieldComponent.new(field: identifier, label_html_attributes: { class: 'col-md-3' }, value_html_attributes: { class: 'col-md-9' }) %>
 <% end %>
 
-<% document.mods.location.each do |location| %>
+<% document.mods.location.reject { |location| location.values.detect { |loc| /#{Settings.PURL_EMBED_RESOURCE}#{document.druid}/.match?(loc) } }.each do |location| %>
```

Comment:
Can you move this into a component or a helper?  We should try to keep logic like this out of the ERBs.  Also please add a comment about what it's doing. It's not immediately clear.